projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -912,6 +909,37 @@ function get_current_url() {
 	return $current_url;
 }
 
+/**
+ * Get the submit button text based on the subscription status.
+ *
+ * @param array $data Array containing block view data.
+ *
+ * @return string
+ */
+function get_submit_button_text( $data ) {
+	if ( ! Jetpack_Memberships::is_current_user_subscribed() ) {
+		return sanitize_submit_text( $data['submit_button_text'] );
```

Comment:
I think it would be fine to keep the sanitization closer to output.

Reply:
good call